conans/client/generators/msvc.py

```diff
@@ -0,0 +1,208 @@
+          </PropertyGroup>
+          <PropertyGroup>
+            <LocalDebuggerEnvironment>PATH=%PATH%;$(Conan{name}BinaryDirectories)
+            </LocalDebuggerEnvironment>
+            <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
+          </PropertyGroup>
+          <ItemDefinitionGroup>
+            <ClCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)
+              %(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ClCompile>
+            <Link>
+              <AdditionalLibraryDirectories>$(Conan{name}LibraryDirectories)
+              %(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+              <AdditionalDependencies>$(Conan{name}Libraries)%(AdditionalDependencies)
+              </AdditionalDependencies>
+              <AdditionalDependencies>$(Conan{name}SystemDeps)%(AdditionalDependencies)
+              </AdditionalDependencies>
+              <AdditionalOptions>$(Conan{name}LinkerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </Link>
+            <Midl>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+            </Midl>
+            <ResourceCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)
+              %(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ResourceCompile>
+          </ItemDefinitionGroup>
+          <ItemGroup />
+        </Project>
+        """)
+
+    @property
+    def filename(self):
+        return None
+
+    @ staticmethod
+    def _name_condition(settings):
+        toolset = msvs_toolset(settings)
+        if toolset is None:
+            raise ConanException("Undefined Visual Studio version %s" %
+                                 settings.get_safe("compiler.version"))
+
+        props = [("Configuration", settings.build_type),
+                 ("Platform", {'x86': 'Win32',
+                               'x86_64': 'x64'}.get(settings.get_safe("arch"))),
+                 ("PlatformToolset", toolset)]
+
+        name = "".join("_%s" % v for _, v in props if v)
+        condition = " And ".join("'$(%s)' == '%s'" % (k, v) for k, v in props if v)
+        return name.lower(), condition
+
+    def _multi(self, name_multi, name_conf, condition):
+        # read the existing mult_filename or use the template if it doesn't exist
```

Comment:
I really don't like reading and modifying files, it is against the generator paradigm we have until now where all the information for the generator is provided by Conan and all files could be generated using a template

Reply:
This is not new. The ``visual_studio_multi`` generator has been doing this for a very long time. It is the only way to avoid load errors, or you need to ``conan install`` all the configurations before you can open the IDE, and that was very annoying. Furthermore, the use case here was more dynamic, in ``cmake_multi`` only the ``build_type`` is involved, but here there were requested changes based on the toolset too.

Reply:
OMG, VS `multi` generators are evil then

Reply:
Yes, a bit, but on the good side, it works surprisingly well, because it is XML, so we haven't had big issues. I'd say that it works more smoothly than the ``cmake_multi`` one.